homeassistant/components/awair/sensor.py

```diff
@@ -116,7 +116,11 @@ def unique_id(self) -> str:
         # sensor for first-gen Awair devices, but its unique_id reflected the truth:
         # under the hood, it was a "DUST" sensor. So we preserve that specific unique_id
         # for users with first-gen devices that are upgrading.
-        if self._kind == API_PM25 and API_DUST in self._air_data.sensors:
+        if (
+            self._kind == API_PM25
+            and self._air_data
+            and API_DUST in self._air_data.sensors
+        ):
```

Comment:
Not sure if this change is OK?

Reply:
`self._air_data` is never set in the init methods of the entity.

Reply:
Nevermind, I see it's a property.

Reply:
This looks ok.